substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ClassEntry.java

```diff
@@ -335,10 +337,9 @@ public ClassEntry getSuperClass() {
         return superClass;
     }
 
-    public Range makePrimaryRange(String methodName, String symbolName, String paramSignature, String returnTypeName, StringTable stringTable, FileEntry primaryFileEntry, int lo,
-                    int hi, int primaryLine,
-                    int modifiers, boolean isDeoptTarget) {
-        FileEntry fileEntryToUse = primaryFileEntry;
+    public Range makePrimaryRange(String methodName, String symbolName, String paramSignature, String returnTypeName, StringTable stringTable, MethodEntry method, int lo,
+                                  int hi, int primaryLine, boolean isDeoptTarget) {
+        FileEntry fileEntryToUse = method.fileEntry;
         if (fileEntryToUse == null) {
             /*
              * Search for a matching method to supply the file entry or failing that use the one
```

Comment:
So this iteration through the methods is now no longer needed because you have already done it under ensureMethodEntry.

Reply:
Correct. My plan is to fix this with https://github.com/oracle/graal/pull/3290

Reply:
Well, you should really delete the loop and comment as part of this commit since your existign changes have already made it redundant.

Reply:
OK, done in https://github.com/oracle/graal/pull/3304/commits/55503f9cc20fe2a80ddbe5adee0f839450eea777